src/lib/fetcher.ts

```diff
@@ -45,7 +53,47 @@ export default class Fetcher {
     return contentTypes
   }
 
-  async getLocalesForSpace () {
+  async getEditorInterfacesInIntents (intentList: IntentList): Promise<Map<string, APIEditorInterfaces>> {
+    const contentTypeIds: string[] = _.uniq(
+      intentList.getIntents()
+        .filter((intent) => intent.isEditorInterfaceUpdate())
+        .reduce((ids, intent) => {
+          const intentIds = intent.getRelatedContentTypeIds()
+          return ids.concat(intentIds)
+        }, [])
+    )
+
+    let editorInterfaces = new Map<string, APIEditorInterfaces>()
+
+    if (contentTypeIds.length === 0) {
+      return editorInterfaces
+    }
+
+    for (let id of contentTypeIds) {
+      try {
+        const response = await this.makeRequest({
```

Comment:
For readability we should move this to it's own method, as this is the only place where we currently allow resources to not exist, which makes the code a bit bloated.

But correct approach 👍